File: Php/app/Urls/BaseClasses/HomegameUrlModel.php
namespace app\Urls\BaseClasses{

	use app\UrlFormatter;
	use entities\Homegame;

	class HomegameUrlModel extends UrlModel{

		public function __construct($format, Homegame $homegame){
			$url = UrlFormatter::formatHomegame($format, $homegame);
			parent::__construct($url);
		}

	}

}

File: Web/Models/Url/HomegameUrlModel.cs
using Core.Classes;

namespace Web.Models.Url{

	public class HomegameUrlModel : UrlModel{

	    public HomegameUrlModel(string format, Homegame homegame) : base(UrlFormatter.FormatHomegame(format, homegame))
	    {
	    }

	}

}